### ISSUE TITLE
False positive MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR in constructor lambda

### ISSUE THREAD

**Issue by davidburstrom** [spotbugs/spotbugs#1867]
Given SpotBugs 4.5.1 and JDK 1.8, the following snippet gets a false positive related to the `StringBuilder.apply` method:

```java
package example;

import java.util.function.Function;

public class Example {

  private final Function<String, String> function;

  public Example() {
    // MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR false positive: Claims "append" is overridable
    function = v -> v + "!";

    // just to ignore warnings about unused member
    System.out.println(function.apply("hello"));
  }
}
```

The attached zip-file is a complete reproducer:

[spotbugs-bug-overridable-method.zip](https://github.com/spotbugs/spotbugs/files/7703365/spotbugs-bug-overridable-method.zip)

**Comment by ThrawnCA** [comment]
Lambda functions are not handled correctly; this is known.

**Comment by JuditKnoll** [comment]
As far as I can see, this was fixed by https://github.com/spotbugs/spotbugs/pull/2900.

**Comment by davidburstrom** [comment]
Thank you @JuditKnoll, I've tried it out locally and can confirm that the bug is fixed.

**Comment by romani** [comment]
Unfortunately it didn't not fix our issue.

I tried to remove suppression https://github.com/checkstyle/checkstyle/pull/15574
And problem still there, should we create new issue?

**Comment by davidburstrom** [comment]
I don't have any mandate here, but I'd reckon you would be able to create a new issue on SpotBugs with a minimal reproducer based on your Checkstyle code.

**Comment by romani** [comment]
Looks like if method is protected final , it is still violation by findbugs, it is still defect on findbug side I think.

Workaround is to make class as final.
